File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder answer = new StringBuilder(); //메모리초과 이슈로 String.concat() 대신 StringBuilder.append()

        Stack<Integer> my_stack = new Stack<Integer>();
        int push_max;
        int rest_target = push_max = sc.nextInt();
        int target = sc.nextInt();  rest_target--;  //새 타겟을 받을때마다 감소

        for(int i=1; i<=push_max; i++) {
            my_stack.push(i);
            answer.append("+\n");

            while( !my_stack.empty() && my_stack.peek()==target ) {
                my_stack.pop();
                answer.append("-\n");
                
                if(rest_target-- > 0)
                    target = sc.nextInt();
            }
        }

        if(my_stack.empty()) System.out.println(answer);
        else                 System.out.println("NO");
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
push_max = int(input())
answer = []

my_stack = []
rest_target = push_max
target = int(input())
rest_target -= 1

for i in range(1, push_max + 1):
    my_stack.append(i)
    answer.append("+\n")

    while my_stack and my_stack[-1] == target:
        my_stack.pop()
        answer.append("-\n")
        
        if rest_target > 0:
            target = int(input())
            rest_target -= 1

if not my_stack:
    print("".join(answer))
else:
    print("NO")